docs/docs/debugging-replace-renderer-api.md

```diff
@@ -0,0 +1,113 @@
+---
+title: Debugging replaceRenderer API
+---
+
+## What is the `replaceRenderer` API?
+
+The `replaceRenderer` API is one of [Gatsby's Server Side Rendering (SSR) hooks](https://www.gatsbyjs.org/docs/ssr-apis/#replaceRenderer). It's used to customise how Gatsby renders your static content. It can be implemented by any Gatsby plugin or your `gatsby-ssr.js` file - adding support for Redux, CSS-in-JS libraries or any code that needs to change Gatsby's default HTML output.
+
+## Why does it cause build errors?
+
+When using `replaceRenderer` many times in your project, only the newest instance will be used - which can cause unexpected problems with your site.
+
+Note that it's only used during `gatsby build`. It won't cause problems as you build your site with `gatsby develop`.
+
+If your project uses replaceRenderer more than once, gatsby build will warn you:
+
+```
+The "replaceRenderer" api has been implemented multiple times. Only the last implementation will be used.
+This is probably an error, see https://example.com for workarounds.
+Check the following files for "replaceRenderer" implementations:
+/path/to/my/site/node_modules/gatsby-plugin-styled-components/gatsby-ssr.js
```

Comment:
I'd list the plugin name first (as that's what will be recognizable" and then list the exact path to the file implementing the API.

Reply:
What's the best way to get the plugin name? Just extract it from the path?

Reply:
👍